components/app.js

```diff
@@ -0,0 +1,62 @@
+var GIPHY_API_URL = 'https://api.giphy.com';
+var GIPHY_PUB_KEY = 'PSrjuieMUbB6xaWavXNEEZUpKPAYJqcx'
+App = React.createClass({
+    getInitialState() {
+        return {
+            loading: false,
+            searchingText: '',
+            gif: {}
+        };
+    },
+
+    handleSearch: function (searchingText) {
+        this.setState({
+            loading: true
+        });
+        this.getGif(searchingText, function (gif) {
+            this.setState({
+                loading: false,
```

Comment:
zmieniasz `loading` na `false` tylko jak się wszystko uda

jeśli call do serwera się z jakichś powodów nie uda - to circle będzie się kręcił w nieskończoność - na pewno kojarzysz temat od strony użytkownika - nie dołączajmy do grupy developerów, którzy tego nie biorą pod uwagę i zostawiają użytkownika w nadziei, że coś się wczytuje... gdy tak naprawdę już dawno przestało ze względu na jakiś błąd :P